### PR TITLE
Feature/2816 token detection mvp

### PR DESCRIPTION
**Pull request by Cal-L**
**Description**

- Replace contract metadata (static JSON token list) with new TokenListController dynamic list
- Upgrade a few libraries such as Redux + lock versions to guarantee versions even without yarn.lock
- Add warning to the top of custom token page
- Update copy + layout for footer on Wallet page (Home page). Also removed Buy ETH button on wallet footer.
- Add token detection toggle on Security & Privacy page (hooked up to PreferencesController)
- A little bit of refactor (Convert JSX to TSX)

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**
#2816

**Next Steps**

- Needs translation

### CODE REVIEW

File: app/components/UI/Tokens/index.js [hunk 1]
```diff
@@ -173,10 +163,13 @@ class Tokens extends PureComponent {
...
 			tokenBalances,
 			tokenExchangeRates,
-			primaryCurrency
+			primaryCurrency,
+			tokenList
 		} = this.props;
 		const itemAddress = safeToChecksumAddress(asset.address);
-		const logo = asset.logo || ((contractMap[itemAddress] && contractMap[itemAddress].logo) || undefined);
+		const logo = (tokenList[itemAddress] && tokenList[itemAddress].iconUrl) || undefined;
+		// Old way of getting image is no longer needed
```
Comment (sethkfman) [inline app/components/UI/Tokens/index.js:171]: Remove comment prior to completion

File: app/components/UI/Tokens/index.js [hunk 2]
```diff
@@ -235,32 +228,6 @@ class Tokens extends PureComponent {
 		});
 	};
 
-	renderBuyEth() {
```
Comment (sethkfman) [inline app/components/UI/Tokens/index.js:238]: Is there a reason why this is being removed?

File: package.json [hunk 3]
```diff
@@ -195,12 +195,16 @@
...
     "react-native-video": "^5.1.1",
     "react-native-view-shot": "^3.1.2",
-    "react-native-webview": "^11.0.2",
-    "react-redux": "5.1.1",
+    "react-native-webview": "11.0.2",
+    "react-navigation": "^4.4.3",
+    "react-navigation-drawer": "1.4.0",
+    "react-navigation-stack": "1.7.3",
+    "react-navigation-tabs": "2.5.5",
+    "react-redux": "7.2.4",
```
Comment (sethkfman) [inline package.json:203]: This is a pretty big bump. Any concerns with breaking changes between 5.1 -> 7.2?

File: package.json [hunk 4]
```diff
@@ -195,12 +195,16 @@
...
     "readable-stream": "1.0.33",
-    "redux": "4.0.1",
-    "redux-mock-store": "1.5.3",
-    "redux-persist": "5.10.0",
+    "redux": "4.1.0",
+    "redux-mock-store": "1.5.4",
+    "redux-persist": "6.0.0",
```
Comment (sethkfman) [inline package.json:207]: Same question on 5.10 -> 6

File: app/util/tokens.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { Token } from '@metamask/controllers';
+
+/**
+ * Convert token list object to token list array
+ */
+export const tokenListToArray = (tokenList: { [key: string]: Token }) => {
```
Comment (sethkfman) [inline app/util/tokens.ts:6]: Would you mind adding tests for this util?

File: app/components/UI/Tokens/index.js [hunk 6]
```diff
@@ -173,10 +163,13 @@ class Tokens extends PureComponent {
...
 			tokenExchangeRates,
-			primaryCurrency
+			primaryCurrency,
+			tokenList
 		} = this.props;
 		const itemAddress = safeToChecksumAddress(asset.address);
-		const logo = asset.logo || ((contractMap[itemAddress] && contractMap[itemAddress].logo) || undefined);
+		const logo = (tokenList[itemAddress] && tokenList[itemAddress].iconUrl) || undefined;
+		// Old way of getting image is no longer needed
+		// const logo = asset.logo || ((tokenList[itemAddress] && tokenList[itemAddress].logo) || undefined);
```
Comment (gantunesr) [inline app/components/UI/Tokens/index.js:172]: Is this comment necessary?

**Review by sethkfman — COMMENTED** [review]
Added some comments and small updates. Let me know if you have questions

File: app/components/UI/Tokens/index.js (hunk 2 above)
Comment (Cal-L) [reply app/components/UI/Tokens/index.js:238]: This was intentional. This footer button was redundant and part of the UI change.

**Review by Cal-L — COMMENTED** [review]
@sethkfman Added test

**Review by cortisiko — APPROVED** [review]
🚀 🚀 🚀 🚀 🚀

**Pull request merged**